### PR TITLE
8303 task(chore): Upgrade to node 15, preliminary work

### PR DESCRIPTION
**Pull request by creido-welly**
## Context
Dependency updates and general tidy up prior to work needed to upgrade to node 15, encompassing the following:

* Replaced deprecated dependencies for babel-eslint and rollup
* replaced storybook webpack config with `webpackFinal` object property
* Removed legacy pure-react-carousel dependency (gallery components have been moved to corporate-react)
* Removed extraneous @types/classnames (classnames has its own type definitions built in)
* Linting refactor and fixes
* Upgraded dependencies to fix TS conflicts (TypeScript and types for React and React-DOM)

### CODE REVIEW

File: src/assets/styles/20-tools/_mixins/_grid.scss [hunk 1]
```diff
@@ -172,118 +172,6 @@
   justify-self: $value;
 }
 
-/**
```
Comment (creido-welly) [inline src/assets/styles/20-tools/_mixins/_grid.scss:175]: Not used anywhere and for some reason this now causes sass errors when starting up Storybook

File: src/components/FormFieldError/FormFieldError.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import VisuallyHidden from 'VisuallyHidden';
 
 type FormFieldErrorProps = {
   className?: string;
-  errors: string | {};
+  errors: string | Record<string, unknown>;
```
Comment (creido-welly) [inline src/components/FormFieldError/FormFieldError.tsx:9]: Recommended by @typescript-eslint

File: src/components/SearchForm/SearchForm.tsx [hunk 3]
```diff
@@ -10,7 +10,9 @@ type SearchFormProps = {
   children?: React.ReactNode;
   className?: string;
   handleInputChange?: (event: React.FormEvent<HTMLInputElement>) => void;
-  handleSubmit?: (event: React.FormEvent<HTMLFormElement>) => {};
+  handleSubmit?: (
```
Comment (creido-welly) [inline src/components/SearchForm/SearchForm.tsx:13]: Recommended by @typescript-eslint

**Comment by skibinska** [comment]
I am trying to link CC to CR and I am getting the error below. Not sure if I am missing anything.

Screenshot 2021-09-20 at 12 48 20

**Comment by creido-welly** [comment]
> I am trying to link CC to CR and I am getting the error below. Not sure if I am missing anything.

Hmm, investigating 👀

**Comment by creido-welly** [comment]
@skibinska I've fixed a typo which prevented the modules from being bundled - sorry, should have tested it first. Try again now

**Review by skibinska — APPROVED** [review]
I don't fully understand the updates, but tested it locally and it all seems to be working fine. 

I tested linking CC to CR with adding some changes and running storybook.

**Pull request merged**